### PR TITLE
MRG: Faster topo plotting

### PR DESCRIPTION
**Pull request by Eric89GXL**
This uses collections and a single axis to speed up topo plottting. So far it took a simple Neuromag example from ~12 sec to ~2 sec on my machine.

Todo:

* [x] Make all code using `_plot_timeseries` work properly.
* [ ] ~~Collection-ize data plot?~~
* [ ] ~~Convert other `_plot_*` functions if necessary.~~
* [x] Get interactivity working again.

BTW we should also look into using collections for raw.plot. It could speed it up. I plan to do it soon.

### CODE REVIEW

File: mne/viz/topo.py [hunk 1]
```diff
@@ -211,26 +255,34 @@ def _imshow_tfr(ax, ch_idx, tmin, tmax, vmin, vmax, onselect, ylim=None,
 def _plot_timeseries(ax, ch_idx, tmin, tmax, vmin, vmax, ylim, data, color,
                      times, vline=None, x_label=None, y_label=None,
                      colorbar=False):
-    """ Aux function to show time series on topo """
+    """Aux function to show time series on topo using """
```
Comment (jona-sassenhagen) [inline mne/viz/topo.py]: Using what?

**Comment by jona-sassenhagen** [comment]
Oh that's great. Does it speed up CIs?

**Comment by Eric89GXL** [comment]
It should speed up plotting tests eventually yes

**Comment by Eric89GXL** [comment]
Ready for review/merge from my end.

File: mne/viz/topo.py [hunk 2]
```diff
@@ -61,14 +62,32 @@ def iter_topography(info, layout=None, on_pick=None, fig=None,
...
     """
-    import matplotlib.pyplot as plt
+    return _iter_topography(info, layout, on_pick, fig, fig_facecolor,
+                            axis_facecolor, axis_spinecolor, layout_scale)
+
+
+def _iter_topography(info, layout, on_pick, fig,
+                     fig_facecolor='k', axis_facecolor='k',
+                     axis_spinecolor='k', layout_scale=None,
+                     unified=False):
```
Comment (kingjr) [inline mne/viz/topo.py]: lines wrap

File: mne/viz/topo.py [hunk 3]
```diff
@@ -235,9 +302,37 @@ def _plot_timeseries(ax, ch_idx, tmin, tmax, vmin, vmax, ylim, data, color,
...
 
 
-def _check_vlim(vlim):
-    """AUX function"""
-    return not np.isscalar(vlim) and vlim is not None
+def _plot_timeseries_unified(bn, ch_idx, tmin, tmax, vmin, vmax, ylim, data,
+                             color, times, vline=None, x_label=None,
+                             y_label=None, colorbar=False):
+    """Aux function to show multiple time series on topo using a single axes"""
+    import matplotlib.pyplot as plt
```
Comment (kingjr) [inline mne/viz/topo.py]: shouldn't this be at the top?

Comment (Eric89GXL) [inline mne/viz/topo.py]: No we nest all mpl imports to make a weak dependency

Comment (kingjr) [inline mne/viz/topo.py]: ok

File: mne/viz/topo.py (hunk 2 above)
Comment (Eric89GXL) [inline mne/viz/topo.py]: what do you mean?

Comment (kingjr) [inline mne/viz/topo.py]: Can't you have one less line?, e.g. `unified=False` after `fig,`

Comment (Eric89GXL) [inline mne/viz/topo.py]: possibly, but then the order of the arguments wouldn't match with `iter_topography` above, which would I think would hurt readability more than saving a line

Comment (kingjr) [inline mne/viz/topo.py]: ok

**Comment by kingjr** [comment]
LGTM

**Comment by agramfort** [comment]
works like a charm. It's lightning fast now !

thx @Eric89GXL

**Comment by Eric89GXL** [comment]
BTW I only changed evoked topo plots, but the same treatment could be given
to the TFR ones if someone is motivated to do it. The example with them is
quite slow on my system but I haven't checked to see if plotting is the
problem or actual computations.

**Comment by agramfort** [comment]
open an issue so we don't forget

**Comment by jaeilepp** [comment]
I have a feeling that plotting is taking longer than the computations. I can take a look next week.

**Comment by Eric89GXL** [comment]
I suspect that will be the case, too. I actually started doing the TFR plots but decided it would be too much at once. I'll open an issue with pointers.

**Pull request merged**